nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/PutHDFS.java

```diff
@@ -366,14 +374,18 @@ public Object run() {
 
                     // Write FlowFile to temp file on HDFS
                     final StopWatch stopWatch = new StopWatch(true);
-                    session.read(putFlowFile, in -> {
-                        OutputStream fos = null;
-                        Path createdFile = null;
-                        try {
-                            if (conflictResponse.equals(APPEND_RESOLUTION) && destinationExists) {
-                                fos = hdfs.append(copyFile, bufferSize);
-                            } else {
-                                final EnumSet<CreateFlag> cflags = EnumSet.of(CreateFlag.CREATE, CreateFlag.OVERWRITE);
+                    final ResourceTransferSource resourceTransferSource = ResourceTransferSource.valueOf(
+                            context.getProperty(RESOURCE_TRANSFER_SOURCE).getValue());
```

Comment:
The preferred way the read a property containing allowable/described value:
```suggestion
                    final ResourceTransferSource resourceTransferSource = context.getProperty(RESOURCE_TRANSFER_SOURCE).asAllowableValue(ResourceTransferSource.class);
```
```suggestion
                    final ResourceTransferSource resourceTransferSource = ResourceTransferSource.valueOf(
                            context.getProperty(RESOURCE_TRANSFER_SOURCE).getValue());
```